### PR TITLE
fix(pod, lint): fix scopelint errors in pod package

### PR DESCRIPTION
**Pull request by shubham14bajpai**
- pinned variable(s) to avoid scopelint errors
 - fixed UT to test `WithAPIObjects` function

Signed-off-by: shubham <shubham.bajpai@mayadata.io>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: pkg/kubernetes/pod/v1alpha1/pod_test.go [hunk 1]
```diff
@@ -31,6 +31,16 @@ func fakeRunningAPIPodObject(podNames []string) []v1.Pod {
 	return plist
 }
 
+func fakeAPIPodObject(podNames []string) []v1.Pod {
```
Comment (AmitKumarDas) [inline pkg/kubernetes/pod/v1alpha1/pod_test.go]: Can we rename function to `fakeAPIPodList`

Comment (shubham14bajpai) [reply pkg/kubernetes/pod/v1alpha1/pod_test.go]: `fakeAPIPodList` is already used to test `WithAPIList` function

Comment (shubham14bajpai) [reply pkg/kubernetes/pod/v1alpha1/pod_test.go]: Yes we can reuse the `fakeAPIPodList` to test `WithAPIObject`. Should I change that ?

Comment (AmitKumarDas) [reply pkg/kubernetes/pod/v1alpha1/pod_test.go]: yes

Comment (AmitKumarDas) [reply pkg/kubernetes/pod/v1alpha1/pod_test.go]: yes

Comment (shubham14bajpai) [reply pkg/kubernetes/pod/v1alpha1/pod_test.go]: Done sir

**Pull request merged**